[:unicorn:  Uniswap Merge] - imports delight

nenadV91 opened the pull request:
# Summary
Closes our commander Uniswap backmerge task

Updates us to [v4.20.2](https://github.com/Uniswap/interface/releases/tag/v4.20.2)

## Test
**🔭 [GP Swap](https://pr1982--gpswapui.review.gnosisdev.com)**: Gnosis Protocol v2 Swap UI 

## Tasks
- [x] Merge and resolve conflicts
- [x] Fix issues with styles in modals
- [x] Fix issue with missing wallet icons in header
- [x] Implement new NetworkSelector component

## Overview
WIP

elena-zh commented:
Hey @nenadV91 , some UI issues:
1. cannot open networks dropdown in the PR (when not connected to a wallet): 
![image](https://user-images.githubusercontent.com/70885163/145579242-e1da17a7-97fe-460d-94ff-ca25799cd970.png)
2. Networks dropdown in the dark mode: 
![image](https://user-images.githubusercontent.com/70885163/145579659-0b3d7210-89e9-4212-8041-de4a381f70f3.png)
3. Netwoks dropdown also does not look good in the lignt mode
![image](https://user-images.githubusercontent.com/70885163/145579766-ed314df4-441b-4c90-b325-02fe16fdea68.png)
Expected: 
![image](https://user-images.githubusercontent.com/70885163/145579798-97d61b84-0c19-4aff-bf00-06ccd6895e65.png)
4. Tooltips with warnings are missing in the networks dropdown when connected using WC: 
![image](https://user-images.githubusercontent.com/70885163/145580160-2b16359f-58de-4b15-a820-871dc7b499ec.png)
Expected: 
![image](https://user-images.githubusercontent.com/70885163/145580211-1973dfbb-c10d-4599-bd43-b398c84aedff.png)
5. Networks' names blink each 30 sec in the dropdown

elena-zh commented:
I have got a hard error when I switched to xDAI using network switcher
![image](https://user-images.githubusercontent.com/70885163/145581426-b9bf46c9-45ff-4aca-9c5d-86064acff44e.png)

Just to add: there is no Cowswap logo in the hard error page. 
![image](https://user-images.githubusercontent.com/70885163/145581585-1c3a6421-8af5-4627-b8f8-fe28a9f443ad.png)
However, should be
![image](https://user-images.githubusercontent.com/70885163/145581747-c3487e3a-4c9f-4b54-94e8-0fa333f765d1.png)

elena-zh commented:
A few additional issues: 
1. Tooltips look a bit weird on the Profile page
![image](https://user-images.githubusercontent.com/70885163/145584162-8683ca5d-fcfb-4caa-84ee-97f4e91d9442.png)
![image](https://user-images.githubusercontent.com/70885163/145584259-c3007ca8-c683-4d0c-99d3-750b6f15dfe1.png)

However, not only on the profile page: 
![image](https://user-images.githubusercontent.com/70885163/145585063-05555a1e-32f8-47b7-b338-fb0b3ad32567.png)
![image](https://user-images.githubusercontent.com/70885163/145585106-a4aa6277-b29a-414b-ac2c-e5a113ac12e6.png)


2. No numbers in Wrap your ETH section (fix might be somewhere here: https://github.com/gnosis/cowswap/pull/1835#issuecomment-987783317)
![image](https://user-images.githubusercontent.com/70885163/145584559-7799e46b-ea7d-483b-9aa3-867ee6d0584b.png)

3. Unwrap/wrap/approve toast notifications look like an error
![image](https://user-images.githubusercontent.com/70885163/145585379-118692a2-1539-4666-b85b-e417a532f3ca.png)
![image](https://user-images.githubusercontent.com/70885163/145585581-4414196b-4eed-465a-8a17-4d8364a836f8.png)
![image](https://user-images.githubusercontent.com/70885163/145585725-50960c00-2de8-488a-a1fa-166b78a2d3fb.png)

4. order placement toast notification: 
![image](https://user-images.githubusercontent.com/70885163/145585862-4d991cb4-4373-431a-8d66-93600e8faac0.png)
The same toast s displayed when an order is confirmed/expired/cancelled.

nenadV91 commented:
@elena-zh 
1. The same happens in Uniswap
2. Updated
3. Updated
4. On what wallets we want to show this?
5. How to reproduce this?

nenadV91 commented:
@elena-zh `I have got a hard error when I switched to xDAI using network switcher` On which wallet did that happened?

elena-zh commented:
> @elena-zh
> 
> 1. The same happens in Uniswap
> 2. Updated
> 3. Updated
> 4. On what wallets we want to show this?
> 5. How to reproduce this?

@nenadV91 , regarding to p.1. In the current Prod Cowswap version we are able to open the network switcher as not connected users. By opening it, a user sees a list of networks we support. By clicking on any, connection modal is opened. When a connected wallet can support different networks (MM, as an example), we should send a request to switch a network there in accordance with the user's selection before connection. 
I mean, that even if it does not work this way in Uniswap, we still should fix this as this feature is currently working in the CowSwap Prod.

2. Looks better
3. Network dropdown still looks differently in comparison with Prod version (different size, different boarder color, different network indicator color + there is no 'Select a network' dropdown name in the current version)
![image](https://user-images.githubusercontent.com/70885163/145783823-27e008a4-41ab-4f41-86e5-74eea270c51a.png)
![image](https://user-images.githubusercontent.com/70885163/145783900-f8a063e5-f7d1-461b-a294-847c78708cf1.png)

4. Wallets that do not support different networks (TrustWallet, as an example), and those that are connected via WC.
I believe, @W3stside could help you with this as he implemented this feature in #1501 

5. Networks' names blink each 30 sec in the dropdown:
 - Open the app, connect to a wallet
 - Specify a token pair, an amount
 - Open network switcher and trace changes in it

elena-zh commented:
> @elena-zh I have got a hard error when I switched to xDAI using network switcher On which wallet did that happened?

I was connected to MM browser's extension

nenadV91 commented:
@elena-zh My point is does it makes sense to change what Uniswap is doing right now which is if the network switch is not supported for the current wallet, then the dropdown can't be opened. Also I think their UI looks good, are we sure we want to change that?

If I am not mistaken currently only Metamask is supported for network switch from the app.

I've also updated
xDai issue
Tooltips issue
Toast notification issue
'Connect to a wallet' button is changed
Space is missing between balance and Max button in a mobile view
Double USD estimation for Price field

I would need some more clarification regarding
- No numbers in Wrap your ETH section (is this in PR and is it going to be fixed when that PR is merged?)
- order placement toast notification (not sure what this is referring to)

elena-zh commented:
> If I am not mistaken currently only Metamask is supported for network switch from the app.

Unfortunately, you are mistaken. 
Frame wallet supports network switching. ImToken, TokenPocket integrated browsers support network switching (besides MM). 

> @elena-zh My point is does it makes sense to change what Uniswap is doing right now which is if the network switch is not supported for the current wallet, then the dropdown can't be opened. Also I think their UI looks good, are we sure we want to change that?

From my perspective it makes sense, but let ask @anxolin , @biocom , @W3stside opinion on this: 
1. Should we leave it as it is currently implemented in this PR or should we change it how it is working on Prod
![image](https://user-images.githubusercontent.com/70885163/145799672-94d4c4d8-02b3-4e73-b4d3-378d59bc4d60.png)
2. Should we disable dropdown switcher to a not connected user or leave the logic I described [above in the case 1](https://github.com/gnosis/cowswap/pull/1982#issuecomment-992264954)  as it is currently working on Prod? 


Thanks

elena-zh commented:
> I would need some more clarification regarding
> 
>  1.  No numbers in Wrap your ETH section (is this in PR and is it going to be fixed when that PR is merged?)
>  2. order placement toast notification (not sure what this is referring to)

1. I can't know for sure whether it is going to be merged before this Uniswap merge PR or after. May be @W3stside could help to answer this question
2. I'm sorry for such short issues descriptions. So, here you should: 
 - place a sell order
 - place a buy order
 - wait till an order is traded
 - wait till any order is expired
 - cancel an order
 - wrap a token
 - unwrap a token
 - approve a token
And compare all these toast notifications with the current version in Production. You will see that all these confirmation toast notifications are different.

nenadV91 commented:
@elena-zh Regarding `Networks' names blink each 30 sec in the dropdown:`, that happens when you hover over some item right?

elena-zh commented:
@nenadV91 , seems to be that no action is needed for this: just open the dropdown. See the video:
https://watch.screencastify.com/v/fOs442nkGKinqof1Pcam

nenadV91 commented:
@elena-zh Do you maybe have `Disable cache` option checked in the Network tab?

elena-zh commented:
Yes, I have this option checked. 
But I also have it checked everywhere, including our Prod app. And there the networks do not blink...

nenadV91 commented:
Yes but the code implementation changed from the Uniswap side, and the idea is that the component is re-rendering but this image is cached and then there is no blinking, and the regular user will probably not have this option checked. You can see the same thing happens in Uniswap.

elena-zh commented:
@nenadV91 , I can't agree on the point that we should leave everything that comes from Uniswap as it is. 
We have already fixed a lot of issues in our app that came with the code from there, and implemented a lot of enhancements to make our app look better. 
So, if fix of this UI change takes a lot of time and does not worth looking at in this PR, I will create a separate issue for this. Please let me know about it  🙃

nenadV91 commented:
`I can't agree on the point that we should leave everything that comes from Uniswap as it is.` and I agree with you on that point, but the question is, is this actually an issue since by default the cache is enabled. So regular users will not see this happening unless they disable the cache.

anxolin commented:
@nenadV91 in general, as a meta-comment, we should question Uniswap features. We own the app, so we can decide wether we want the new stuff or not. 

In this case, im fine with leaving the new or the old network selector, but the part I wouldn't add is to not allow the users to see the menu

>  is this actually an issue since by default the cache is enabled.

Not a huge deal, however, for me is a red flag. Why is this happening now and not before? there must be some explanation. Previous behaviour was fine. 

In general these effects make the UI look flickering which reduce the trust of the user in it. I'm not so sure we can rely in cache to remove this effect. It seems to me there's some re-rendering or something that cause this and is the real issue

anxolin commented on src/components/Loader/styled.tsx:
```diff
@@ -0,0 +1,39 @@
+import styled, { css, keyframes } from 'styled-components/macro'
+
+export const loadingAnimation = keyframes`
```
it looks like this is a cool loader. Do we use it or want to use it?

anxolin commented on src/components/PrivacyPolicy/index.tsx:
```diff
@@ -0,0 +1,173 @@
+    description: (
+      <Trans>
+        The app securely collects your wallet address and shares it with TRM Labs Inc. for risk and compliance reasons.
+      </Trans>
```
This might be important. Maybe we should do this too? we should try to be compliant

https://github.com/gnosis/cowswap/blob/bbc64f12bb1773bb50c6e343c768a20c91d54ce6/src/hooks/useMonitoringEventCallback.ts#L23


In any case. It's important to at least create an issue, and block in a MOD the leaking of this information.
We should check with legal if this is important to add or not

nenadV91 replied:
Added mod file for now

anxolin commented on src/components/SearchModal/BlockedToken.tsx:
```diff
@@ -0,0 +1,71 @@
+  <Wrapper>
+    <Header>
+      {onBack ? <ArrowLeft style={{ cursor: 'pointer' }} onClick={onBack} /> : <div />}
+      <TYPE.mediumHeader>
```
This components seems like something new we could use? 
Are we using it in this version?

nenadV91 replied:
Yep, this is new modal for blocked tokens https://raw.githubusercontent.com/The-Blockchain-Association/sec-notice-list/master/ba-sec-list.json. Added

![BlockedToken](https://user-images.githubusercontent.com/34926005/146095406-82dc634d-02a2-4ab2-95b6-a31f244d7994.jpeg)

anxolin commented on src/components/SearchModal/TokenImportCard.tsx:
```diff
@@ -0,0 +1,76 @@
+  list?: TokenList
+  token: Token
+}
+const TokenImportCard = ({ list, token }: TokenImportCardProps) => {
```
Is this something new and relevant to us?

nenadV91 replied:
Since the new BlockedToken component is similar to ImportToken component they took that part into this separate component which is now used in both of those components.

anxolin commented on src/components/WalletModal/index.tsx:
```diff
@@ -182,13 +200,18 @@ export default function WalletModal({
+      activate(connector, undefined, true)
+        .then(async () => {
+          const walletAddress = await connector.getAccount()
+          logMonitoringEvent({ walletAddress })
```
Should we add this?  https://github.com/gnosis/cowswap/blob/e4ad97cdba5b3b3f1481fdf38f1e9f32f5e0be11/src/custom/components/WalletModal/WalletModalMod.tsx#L214

we should not leak info to them, but maybwe we want to make use of some of these log/monitoring features

nenadV91 replied:
Since for this we have Mod file this component , this isn't used but I will add this code to mod and comment it out for now.

anxolin commented on src/constants/chains.ts:
```diff
@@ -63,65 +74,89 @@ export type ChainInfo = { readonly [chainId: number]: L1ChainInfo | L2ChainInfo
 
 export const CHAIN_INFO: ChainInfo = {
   [SupportedChainId.ARBITRUM_ONE]: {
+    blockWaitMsBeforeWarning: ms`10m`,
```
Does this need to add XDAI?

nenadV91 replied:
So the default value is `ms`10m`` but we can change this, what do you think it should be?

nenadV91 commented on .github/workflows/check-pr-title.yaml:
```diff
@@ -0,0 +1,17 @@
+    name: Check PR Title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v3.4.0
```
It was auto added, its removed now.

nenadV91 commented on .github/workflows/crowdin-sync.yaml:
```diff
@@ -0,0 +1,57 @@
+# hourly we sync translations from Crowdin
+on:
+  schedule:
+    - cron: '0 * * * *' # every hour we download translations and update the pr from crowdin
```
Removed for now

nenadV91 commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,50 @@
+      - main
+
+jobs:
+  integration-tests:
```
Removed for now

nenadV91 commented on .github/workflows/unit-tests.yaml:
```diff
@@ -0,0 +1,40 @@
+
+jobs:
+  unit-tests:
+    name: Unit tests
```
Removed for now

W3stside commented:
@nenadV91 nice, very nice work! huge apologies for the last merge (i swear i didnt do it on purpose to get you familiar with the entire app + modding) 

everything so far looks good to me:
1. network switcher - seems like the last one, minus the disconnected wallet feature as discussed above. I am in favour of keeping our old implementation
2. modals looks good
3. responsive looks good
will keep checking

elena-zh commented:
Hey @nenadV91 , I did the 2nd round of testing the changes. Most issues are fixed, great job! 
But still, here are some 'old' and new issues: 
1. Wrap/Unwrap in xDAI is recognized as Swap transaction
![swap](https://user-images.githubusercontent.com/70885163/146194308-032fa7de-1f16-44fb-a986-f02f1450aa9e.jpg)
2. Besides, it shows Fee on wrap
![fee request on wrap](https://user-images.githubusercontent.com/70885163/146194421-4a90bdd2-a555-4b58-907a-5b4b97ab074b.jpg)
3. There is no Wrap you XDAI section when XDAI is selected as a Sell token
![wrap my xdai section is missing](https://user-images.githubusercontent.com/70885163/146194718-97d288ce-564a-4df5-a8ec-e78cb921e08a.jpg)
4. The app seems to be disconnected when switch a network in MM injected browser (iOS). So it would be great to force the app reloading after a network switching, like in works in Android. 

https://user-images.githubusercontent.com/70885163/146195203-55fb7b21-bb31-47ad-b60d-3a4aeed51203.mp4

5. Then again, it would be nice to have the network selector enabled when a user is not connected
6. 'Wallet' is the 'Connect to a wallet' button should start with a lowercase letter
![connect to a wallet -](https://user-images.githubusercontent.com/70885163/146195564-ee9f88af-59e6-4450-b228-06395a78d874.jpg)
7. App header overlaps logo on mobile screens with less than 340 px width. 
![less than 350 px](https://user-images.githubusercontent.com/70885163/146196015-6d2dccdf-2196-4579-9f6b-c1fdbd006356.jpg)


Please, let me know if I need to create separate tasks on any of these issues. 

Thanks

alfetopito commented on src/components/Logo/index.tsx:
```diff
@@ -5,7 +5,7 @@ import useTheme from '../../hooks/useTheme'
 const BAD_SRCS: { [tokenAddress: string]: true } = {}
 
-export interface LogoProps extends Pick<ImageProps, 'style' | 'alt' | 'className'> {
+interface LogoProps extends Pick<ImageProps, 'style' | 'alt' | 'className'> {
```
Maybe we want to keep the export as it could have been added by us?

alfetopito commented on src/components/Popups/PopupItem.tsx:
```diff
@@ -1,13 +1,14 @@
+import { PopupContent } from 'state/application/reducer'
 import TransactionPopup from './TransactionPopup'
 
-export const StyledClose = styled(X)`
```
Exports should probably be kept

alfetopito commented on src/components/PositionList/index.tsx:
```diff
@@ -34,7 +34,7 @@ const MobileHeader = styled.div`
   }
 `
 
-export type PositionListProps = React.PropsWithChildren<{
```
Maybe keep the export

alfetopito commented on src/components/PositionListItem/index.tsx:
```diff
@@ -126,7 +127,7 @@ const DataText = styled.div`
   `};
 `
 
-export interface PositionListItemProps {
```
Maybe keep the export

alfetopito commented on src/components/SearchModal/CurrencyList.tsx:
```diff
@@ -1,27 +1,28 @@
 import { MenuItem } from './styleds'
-import Loader from '../Loader'
-import { isTokenOnList } from '../../utils'
-import ImportRow from 'components/SearchModal/ImportRow'
```
I forgot to notice previous files until here, but keep an eye out on the imports.
It might be overwriting our custom changes to make the file moddable.

For example, it's changing `ImportRow` from absolute to relative import.
Likely we changed that to use our custom `ImportRow` implementation.

alfetopito commented on src/components/SearchModal/ManageTokens.tsx:
```diff
@@ -1,22 +1,22 @@
-import useTheme from '../../hooks/useTheme'
-import { Trans } from '@lingui/macro'
 
+import useTheme from '../../hooks/useTheme'
+import { ExplorerDataType, getExplorerLink } from '../../utils/getExplorerLink'
 import { CurrencyModalView } from './CurrencySearchModal'
+import ImportRow from './ImportRow'
```
ImportRow import

alfetopito commented on src/components/Toggle/ListToggle.tsx:
```diff
@@ -30,7 +30,7 @@ const StatusText = styled(TYPE.main)<{ isActive?: boolean }>`
 `
 
-export interface ToggleProps {
+interface ToggleProps {
```
Probably keep the export?

alfetopito reviewed (commented):
Reviewed files from 1 to 101

Files of interest / may need to check the modded versions if any

- dependabot for token lists
- accountdetails/transaction.tsx
- accountdetails/transactionsummary.tsx (new?)
- blocklist/index
- buttom/index
- currencyinputpannel/index 
    - new steylednumericalinput
    - loading indicator for fiat
- currencylogo/index
- header/networkselector
- menu/index
- popups/trnasactionpopup
- searchmodal/currecnysearchmodal
- searchmodal/importtoken
- searchmodal/managetokens
- tooltip/index

alfetopito reviewed (commented):
Reviewed files from 101 to 201

Files that might need further checking 

- web3reactmanager/index
- swap/advancedswapdetails
- swap/tradeprice
- swap/styleds
- constants/chains
- header/networkcard/networkcardmod
- header/networkcard/index

alfetopito commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -85,7 +85,7 @@ import { formatSmart } from 'utils/format'
 import usePrevious from 'hooks/usePrevious'
 import { StyledAppBody } from './styleds'
-import { ApplicationModal } from 'state/application/actions'
+import { ApplicationModal } from '@src/state/application/reducer'
```
Do you have to import from `@src`? In other places you didn't
![Uploading Screenshot from 2021-12-16 14-05-03.png…]()

alfetopito commented on src/custom/state/swap/hooks.ts:
```diff
@@ -37,7 +38,7 @@ import TradeGp from './TradeGp'
 import { WETH9_EXTENDED as WETH, GpEther as ETHER } from 'constants/tokens'
 
-import { BAD_RECIPIENT_ADDRESSES } from 'state/swap/hooks'
+import { BAD_RECIPIENT_ADDRESSES } from '@src/state/swap/hooks'
```
Do you have to use `@src`?

alfetopito commented on src/custom/state/swap/hooks.ts:
```diff
@@ -422,10 +428,10 @@ export function queryParametersToSwapState(parsedQs: ParsedQs, defaultInputCurre
   return {
     [Field.INPUT]: {
-      currencyId: inputCurrency,
+      currencyId: inputCurrency === '' ? null : inputCurrency ?? null,
```
I understand this is coming from the merge but, hum?

Am I reading this wrong or is this equivalent to `inputCurrecy || null` ?

W3stside replied:
sure seems loik it

alfetopito commented on src/hooks/useERC20Permit.ts:
```diff
@@ -114,7 +115,7 @@ const PERMIT_ALLOWED_TYPE = [
 ]
 
-export function useERC20Permit(
+function useERC20Permit(
```
Will probably need the export?

alfetopito commented on src/hooks/useFetchListCallback.ts:
```diff
@@ -1,9 +1,9 @@
-import { getNetworkLibrary } from 'connectors'
 import { useAppDispatch } from 'state/hooks'
+
+import { getNetworkLibrary } from '../connectors'
```
Keep absolute import?

alfetopito commented on src/hooks/useIsSwapUnsupported.ts:
```diff
@@ -1,6 +1,7 @@
 import { useMemo } from 'react'
-import { useUnsupportedTokens } from 'hooks/Tokens'
+
+import { useUnsupportedTokens } from './Tokens'
```
Keep absolute import?

alfetopito commented on src/index.tsx:
```diff
@@ -46,32 +45,14 @@ if (!!window.ethereum) {
       <ListsUpdater />
       <UserUpdater />
       <ApplicationUpdater />
+      <TransactionUpdater />
```
Should we keep this updater?
Maybe we already disabled, but should check

alfetopito reviewed (commented):
Reviewed files from 202 to 314

- custom/utils/blocknative - check whether they export their typings
- src/index

alfetopito commented on src/pages/App.tsx:
```diff
@@ -6,10 +6,10 @@ import AddressClaimModal from '../components/claim/AddressClaimModal'
 import Header from '../components/Header'
 import Polling from '../components/Header/Polling'
-import Popups from 'components/Popups'
+import Popups from '../components/Popups'
```
Keep import

alfetopito commented on src/pages/Swap/index.tsx:
```diff
@@ -2,42 +2,51 @@ import { Trans } from '@lingui/macro'
+import TradePrice from 'components/swap/TradePrice'
 import UnsupportedCurrencyFooter from 'components/swap/UnsupportedCurrencyFooter'
-import { MouseoverTooltip, MouseoverTooltipContent } from 'components/Tooltip'
+import { MouseoverTooltip, MouseoverTooltipContent } from '@src/components/Tooltip'
```
Need `@src`?

alfetopito commented on src/pages/Swap/index.tsx:
```diff
@@ -2,42 +2,51 @@ import { Trans } from '@lingui/macro'
+
 import AddressInputPanel from '../../components/AddressInputPanel'
-import { ButtonConfirmed, ButtonError, ButtonGray, ButtonLight, ButtonPrimary } from 'components/Button'
+import { ButtonConfirmed, ButtonError, ButtonLight, ButtonPrimary } from '../../components/Button'
```
Probably keep imports

alfetopito commented on src/state/application/hooks.ts:
```diff
@@ -1,8 +1,10 @@
-import { useActiveWeb3React } from 'hooks/web3'
-import { AppState } from 'state'
-import { addPopup, ApplicationModal, PopupContent, removePopup, setOpenModal } from 'state/application/actions'
+import { useAppDispatch, useAppSelector } from '@src/state/hooks'
```
Need `@src`?

alfetopito commented on src/state/application/hooks.ts:
```diff
@@ -1,8 +1,10 @@
+import { useAppDispatch, useAppSelector } from '@src/state/hooks'
+
+import { useActiveWeb3React } from '../../hooks/web3'
+import { AppState } from '../index'
```
Relative imports

alfetopito commented on src/state/swap/hooks.tsx:
```diff
@@ -1,26 +1,29 @@
-import { isAddress } from 'utils'
-import { AppState } from 'state'
+import { useActiveWeb3React } from '../../hooks/web3'
+import { isAddress } from '../../utils'
```
Check imports

alfetopito commented on src/state/transactions/hooks.tsx:
```diff
@@ -1,38 +1,33 @@
+import { useTransactionMonitoringEventCallback } from 'hooks/useMonitoringEventCallback'
 import { useCallback, useMemo } from 'react'
-import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { useAppDispatch, useAppSelector } from '@src/state/hooks'
```
Need `@src`?

alfetopito commented on src/state/wallet/hooks.ts:
```diff
@@ -1,18 +1,19 @@
+import { Currency, CurrencyAmount, Ether, Token } from '@uniswap/sdk-core'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { SupportedChainId } from '@src/constants/chains'
```
`@src`

alfetopito commented on src/state/wallet/hooks.ts:
```diff
@@ -1,18 +1,19 @@
-import { UNI } from 'constants/tokens'
-import { useActiveWeb3React } from '../../hooks/web3'
+
+import { UNI } from '../../constants/tokens'
```
Import

alfetopito commented on src/theme/components.tsx:
```diff
@@ -220,7 +220,7 @@ const rotate = keyframes`
 `
 
-export const Spinner = styled.img`
+const Spinner = styled.img`
```
Keep export?

alfetopito commented on src/utils/index.ts:
```diff
@@ -3,8 +3,9 @@ import { AddressZero } from '@ethersproject/constants'
-import { TokenAddressMap } from 'state/lists/hooks'
+import { FeeAmount } from '@uniswap/v3-sdk'
+
+import { TokenAddressMap } from '../state/lists/hooks'
```
Import

alfetopito commented on src/utils/retry.ts:
```diff
@@ -9,7 +9,7 @@ function waitRandom(min: number, max: number): Promise<void> {
  * This error is thrown if the function is cancelled before completing
  */
-export class CancelledError extends Error {
+class CancelledError extends Error {
```
Keep export?

alfetopito commented on tsconfig.json:
```diff
@@ -1,44 +1,14 @@
   ],
   "include": [
-    "src"
+    "src/**/*"
```
Do you need this?

alfetopito reviewed (commented):
Checked the final ~100 files

- state/application/actions - make sure behaviour is still needed and captured elsewhere
- state/transactions/actions - new transaction types
- utils/addnetwork - is it useful for us?
- utils/gettokenlists - lazy loading dependencies, we could reuse it for other stuff

nenadV91 commented:
I've adjusted react-cosmos config so its working in our app and also added examples for 2 components. To use this you can run `yarn run cosmos` and it will start a server on http://localhost:5000/ where you can preview the components. Basically to add some component to cosmos you create a __fixtures__ folder or a file with .fixture.tsx name. I don't know if we want to use it or not but now we can and know how.

![react-compass](https://user-images.githubusercontent.com/34926005/146553401-a8aaa664-2423-4ede-934d-463aea02b480.png)

nenadV91 commented on cypress/support/commands.js:
```diff
@@ -4,6 +4,7 @@
 // https://on.cypress.io/custom-commands
 // ***********************************************
 
+import { Eip1193Bridge } from '@ethersproject/experimental/lib/eip1193-bridge'
```
Removed

elena-zh commented:
Hey @nenadV91 , great job!
The vast majority of issues is fixed now. 

Remaining:
1. It would be great to remove 'Auto Router' from the settings in a mobile view
![image](https://user-images.githubusercontent.com/70885163/146775750-04b06f4c-4494-4198-9554-b38bf3eb817c.png)

2. Case with network switching in  mobile MM integrated browser is still here: The app seems to be disconnected when switch a network in MM injected browser. So it would be great to force the app reloading after a network switching. But let me know if I need to create a low-prio separate issue for this

https://user-images.githubusercontent.com/70885163/146195203-55fb7b21-bb31-47ad-b60d-3a4aeed51203.mp4

3. Just to be on the same page for the network switcher. Currently, it works great besides the part of the logic when we are connected to a wallet using WC: the switcher is disabled in this case, Should we leave it as it is? @anxolin , @biocom , @alfetopito ?
![disabled](https://user-images.githubusercontent.com/70885163/146776236-56315cc3-08ee-4402-98bb-4c5751c2fe7f.jpg)

elena-zh commented:
> Case with network switching in mobile MM integrated browser is still here: The app seems to be disconnected when switch a network in MM injected browser. So it would be great to force the app reloading after a network switching. But let me know if I need to create a low-prio separate issue for this

Actually, I have created #2019 bug as it is not only related to MM wallet. Besides, the issue is reproducible on Prod now.

alfetopito commented:
> It would be great to remove 'Auto Router' from the settings in a mobile view

Auto router section should be hidden everywhere as we do not have that option on our side.

nenadV91 commented:
Auto-router is removed now

alfetopito commented on src/components/Popups/PopupItem.tsx:
```diff
@@ -16,7 +17,7 @@ export const StyledClose = styled(X)`
   }
 `
-export const Popup = styled.div`
+const Popup = styled.div`
```
Keep this one also?

alfetopito commented on src/components/TransactionConfirmationModal/index.tsx:
```diff
@@ -1,23 +1,25 @@
-import { useIsTransactionConfirmed, useTransaction } from 'state/transactions/hooks'
-import Badge from 'components/Badge'
+import { useActiveWeb3React } from '../../hooks/web3'
+import { ExternalLink } from '../../theme'
+import { CloseIcon, CustomLightSpinner } from '../../theme/components'
```
Import on these 2 lines

alfetopito commented on src/components/TransactionSettings/index.tsx:
```diff
@@ -1,17 +1,17 @@
+import { darken } from 'polished'
+import { useContext, useState } from 'react'
+import { useSetUserSlippageTolerance, useUserSlippageTolerance, useUserTransactionTTL } from 'state/user/hooks'
 import styled, { ThemeContext } from 'styled-components/macro'
 
-import QuestionHelper from '../QuestionHelper'
-import { TYPE } from 'theme'
+import { TYPE } from '../../theme'
```
import

alfetopito commented on src/components/TransactionSettings/index.tsx:
```diff
@@ -1,17 +1,17 @@
-import { useState, useContext } from 'react'
+import { Trans } from '@lingui/macro'
 import { Percent } from '@uniswap/sdk-core'
+import { L2_CHAIN_IDS } from '@src/constants/chains'
```
`@src`

alfetopito commented on src/custom/components/Identicon/IdenticonMod.tsx:
```diff
@@ -1,41 +1,33 @@
-import Jazzicon from '@metamask/jazzicon'
 import { IdenticonProps } from 'components/Identicon'
 
+import { useActiveWeb3React } from '../../hooks/web3'
```
import

alfetopito commented on src/custom/components/Tooltip/TooltipMod.tsx:
```diff
@@ -16,14 +16,18 @@ export interface TooltipProps extends Omit<PopoverProps, 'content' | 'PopoverCon
   content: ReactNode
+  onOpen?: () => void
+  // whether to wrap the content in a `TooltipContainer`
+  // wrap?: boolean
```
Maybe we could keep this option and set the default to `true` which is the current behaviour

alfetopito commented on src/custom/components/TransactionSettings/TransactionSettingsMod.tsx:
```diff
@@ -1,17 +1,17 @@
-import { useState, useContext } from 'react'
+import { Trans } from '@lingui/macro'
 import { Percent } from '@uniswap/sdk-core'
+import { L2_CHAIN_IDS } from '@src/constants/chains'
```
`@src`

alfetopito commented on src/custom/components/TransactionSettings/TransactionSettingsMod.tsx:
```diff
@@ -245,7 +245,11 @@ export default function TransactionSettings({ placeholderSlippage }: Transaction
-              text={t`Your swap expires and will not execute if it is pending for longer than the selected duration. ${INPUT_OUTPUT_EXPLANATION}`}
+              text={
+                <Trans>
+                  Your swap expires and will not execute if it is pending for longer than the selected duration.
```
You can keep the explanation, it just wouldn't work with `${INPUT_OUTPUT_EXPLANATION}`, but it will with `{INPUT_OUTPUT_EXPLANATION}`

alfetopito commented on src/custom/constants/chains/chainsMod.ts:
```diff
@@ -11,92 +14,172 @@ export enum SupportedChainId {
   }, */
+  [SupportedChainId.XDAI]: {
+    docs: 'https://docs.uniswap.org/',
+    explorer: 'https://gnosis-protocol.io/xdai',
```
I know this was there already, but the explorer should be BlockScout rather than our explorer

alfetopito commented on src/pages/Swap/index.tsx:
```diff
@@ -2,42 +2,51 @@ import { Trans } from '@lingui/macro'
 import { AutoColumn } from '../../components/Column'
 import CurrencyInputPanel from 'components/CurrencyInputPanel'
-import CurrencyLogo from 'components/CurrencyLogo'
+import CurrencyLogo from '../../components/CurrencyLogo'
```
import

alfetopito commented on src/pages/Swap/index.tsx:
```diff
@@ -49,22 +58,21 @@ import {
-import { useExpertModeManager, useUserSingleHopOnly } from 'state/user/hooks'
-import { HideSmall, LinkStyledButton, TYPE } from 'theme'
+import { useExpertModeManager } from '../../state/user/hooks'
+import { LinkStyledButton, TYPE } from '../../theme'
```
import

alfetopito commented on src/pages/Swap/index.tsx:
```diff
@@ -49,22 +58,21 @@ import {
 import { maxAmountSpend } from '../../utils/maxAmountSpend'
 import { warningSeverity } from '../../utils/prices'
-import AppBody from 'pages/AppBody'
+import AppBody from '../AppBody'
```
import

alfetopito commented on src/state/application/hooks.ts:
```diff
@@ -1,8 +1,10 @@
 import { useActiveWeb3React } from 'hooks/web3'
 import { AppState } from 'state'
-import { addPopup, ApplicationModal, PopupContent, removePopup, setOpenModal } from 'state/application/actions'
+import { addPopup, ApplicationModal, PopupContent, removePopup, setOpenModal } from './reducer'
```
import

alfetopito reviewed (commented):
Few more things, ignore if you already checked and they are not needed.

W3stside commented on src/components/AddressInputPanel/index.tsx:
```diff
@@ -1,9 +1,11 @@
 import useENS from '../../hooks/useENS'
 import { useActiveWeb3React } from '../../hooks/web3'
-import { ExternalLink, TYPE } from 'theme'
+import { ExternalLink, TYPE } from '../../theme'
```
path

W3stside commented on src/components/Badge/index.tsx:
```diff
@@ -13,7 +13,7 @@ export enum BadgeVariant {
 }
 
-export interface BadgeProps {
+interface BadgeProps {
```
export?

W3stside commented on src/components/Blocklist/index.tsx:
```diff
@@ -1,15 +1,25 @@
 
 // SDN OFAC addresses
 const BLOCKED_ADDRESSES: string[] = [
+  '0x7Db418b5D567A4e0E8c59Ad71BE1FcE48f3E6107',
```
make a note of these blocked addresses please

W3stside commented on src/components/Header/index.tsx:
```diff
@@ -284,7 +290,7 @@ export default function Header() {
           <Trans>Pool</Trans>
         </StyledNavLink>
-        {chainId && chainId === SupportedChainId.MAINNET && (
+        {(!chainId || chainId === SupportedChainId.MAINNET) && (
```
so no it's if you're not connected or you're on mainnet?

W3stside commented:
after some minor import changes, looks good to me now, @gnosis/gp-qa anything holding this back?

elena-zh commented:
@W3stside , @nenadV91 the latest changes LGTM!
The only thing remains is to agree on this point: 

>  Just to be on the same page for the network switcher. Currently, it works great besides the part of the logic when we are connected to a wallet using WC: the switcher is disabled in this case, Should we leave it as it is? 
> ![disabled](https://user-images.githubusercontent.com/70885163/146776236-56315cc3-08ee-4402-98bb-4c5751c2fe7f.jpg)
> 
> 
>

W3stside reviewed (approved):
Very nice work. Take a break form this for a while 😅
